pkg/operation/botanist/component/logging/fluentoperator/fluent_operator.go

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fluentoperator
+
+import (
+	"context"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	vpaautoscalingv1 "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/utils"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+)
+
+const (
+	// OperatorManagedResourceName is the name of the Fluent Operator managed resource.
+	OperatorManagedResourceName = "fluent-operator"
+	name                        = "fluent-operator"
+	roleName                    = "gardener.cloud:logging:fluent-operator"
+	vpaName                     = "fluent-operator-vpa"
+)
+
+var (
+	protocolTCP = utils.ProtocolPtr(corev1.ProtocolTCP)
+	lokiPort    = utils.IntStrPtrFromInt(3100)
+)
+
+// FluentOperatorValues are the values for the Fluent Operator.
+type FluentOperatorValues struct {
+	// Image of the operator.
+	OperatorImage string
+	// FluentBitImage is the Fluent-bit image
```

Comment:
This doc string sounds wrong

Reply:
This comment is not yet resolved.

```suggestion
	// OperatorPriorityClassName is the name of the priority class of the fluent-operator.
```